### PR TITLE
duck: use `libffi` from macos

### PR DESCRIPTION
**Pull request by carlocab**

### CODE REVIEW

**Comment by carlocab** [comment]
```
dispatch.c:433:8: error: use of undeclared identifier 'FFI_BAD_ARGTYPE'
    case FFI_BAD_ARGTYPE:
         ^
```
Weird.

**Comment by Bo98** [comment]
FFI_BAD_ARGTYPE is available in libffi 3.4 and later, i.e. Monterey or later.

**Comment by carlocab** [comment]
Thanks for the tip!

File: Formula/duck.rb [hunk 1]
```diff
@@ -60,12 +59,18 @@ def install
...
-                  "LIBRARY=$(BUILD)/$(LIBPFX)jnidispatch$(JNISFX)",
-                  "LIBRARY=$(BUILD)/$(LIBPFX)jnidispatch$(LIBSFX)"
+        inreplace "native/Makefile" do |s|
+          libffi_libdir = if MacOS.version >= :monterey
+            MacOS.sdk_path/"usr/lib"
+          else
+            Formula["libffi"].opt_lib
+          end
+          # Add linker flags for libffi because Makefile call to pkg-config doesn't seem to work properly.
+          s.change_make_var! "LIBS", "-L#{libffi_libdir} -lffi"
```
Comment (carlocab) [inline Formula/duck.rb:69]: I wonder if I can scope this `inreplace` to just the versions we are using the `libffi` formula.

**Pull request closed without merging**